lib/internal/validators.js

```diff
@@ -233,9 +229,17 @@ const validateAbortSignal = hideStackFrames((signal, name) => {
   }
 });
 
-const validateFunction = hideStackFrames((value, name) => {
-  if (typeof value !== 'function')
-    throw new ERR_INVALID_ARG_TYPE(name, 'Function', value);
+const validateFunction = hideStackFrames((value, name, allowClass = false) => {
+  if (typeof value === 'function') {
+    if (allowClass !== true) return;
```

Comment:
```suggestion
    if (allowClass) return;
```
This should be the opposite, right? If the validator allows a class then it should be safe to return here but if it doesn't then we should check that the string inspect() returns does not start with "[class".

Reply:
Oh yea my bad, fixed.